src/main/java/com/automationexercise/Pages/HomePage.java

```diff
@@ -0,0 +1,68 @@
+package com.automationexercise.Pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.List;
+
+public class HomePage extends BasePage {
+
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    By ProductButton = By.xpath("//a[@href=\"/products\"]");
+    WebElement AllProductsPage = driver.findElement
```

Comment:
You should save any WebElement object because it will cause an StaleElementRefrence. Just you save the locator as By object and find it whenever you need.
Check this:https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/#stale-element-reference-exception